spec/requests/api/collections_spec.rb

```diff
@@ -27,12 +27,20 @@ def test_collection_bulk_query(collection, collection_url, klass, id = nil)
 
     obj = id.nil? ? klass.first : klass.find(id)
     url = send("#{collection}_url", obj.id)
-    run_post(collection_url, gen_request(:query, [{"id" => obj.id}, {"href" => url}]))
+    attr_list = String(Api::Settings.collections[collection].parsed_by_attrs).split(",")
```

Comment:
I'm not against the other object class, it's the usability, accessibility via [] and .<collection>  so that it's similar to the Api::Settings. (and eliminating the hidden new).  If Api::CollectionConfig can subclass Api::Settings.collections, you'd have [:vms] and .vms access already.